### PR TITLE
Update GitHub theme, add dark mode

### LINKED ISSUE #1906: Github Dark style
I was trying to use [primer/css](https://github.com/primer/css) with markdown and when combined with rouge the code sections work as one would expect in light mode. But when using primer's dark mode (any of them) `rouge`'s highlighting is a little off. Is there any chance you would consider adding a `github.dark` style?

Thanks!

### PR DESCRIPTION
**Pull request by dunkmann00**
This updates the GitHub theme and also adds support for a dark mode.

The color values are sourced from the primer/primitives repository. https://github.com/primer/primitives

This resolves #1906

### CODE REVIEW

File: lib/rouge/themes/github.rb [hunk 1]
```diff
@@ -6,68 +6,139 @@ module Themes
...
+
+      style Error,                      :fg => :fgError, :bg => :bgError
+
+      style Comment,
+            Generic::Lineno,
+            Generic::Traceback,         :fg => :comment
+
+      style Name::Entity,
+            Literal::String::Interpol,
+            Text,                       :fg => :fgDefault
```
Comment (tancnle) [inline lib/rouge/themes/github.rb:136]: **suggestion:** Could we add a background colour to distinguish between light and dark themes? FYI, we [use](https://github.com/rouge-ruby/rouge/blob/master/lib/rouge/theme.rb#L112) styles on `Text` token to set the general foreground and background colour.

```
Text, :fg => :fgDefault, :bg => :bgDefault
```

File: lib/rouge/themes/github.rb [hunk 2]
```diff
@@ -6,68 +6,139 @@ module Themes
...
+        palette :comment     => P_GRAY_3[@mode]
+        palette :constant    => P_BLUE_2[@mode]
+        palette :entity      => P_PURPLE_2[@mode]
+        palette :heading     => P_BLUE_5[@mode]
+        palette :keyword     => P_RED_3[@mode]
+        palette :string      => P_BLUE_1[@mode]
+        palette :tag         => P_GREEN_1[@mode]
+        palette :variable    => P_ORANGE_2[@mode]
+
+        palette :fgDefault   => P_GRAY_1[@mode]
```
Comment (tancnle) [inline lib/rouge/themes/github.rb:57]: **suggestion:** Could we add a `bgDefault` palette for the background colour?

```ruby
palette :bgDefault   => P_GRAY_9[@mode]
```

File: lib/rouge/themes/github.rb [hunk 3]
```diff
@@ -6,68 +6,139 @@ module Themes
...
+        palette :comment     => P_GRAY_5[@mode]
+        palette :constant    => P_BLUE_6[@mode]
+        palette :entity      => P_PURPLE_5[@mode]
+        palette :heading     => P_BLUE_6[@mode]
+        palette :keyword     => P_RED_5[@mode]
+        palette :string      => P_BLUE_8[@mode]
+        palette :tag         => P_GREEN_6[@mode]
+        palette :variable    => P_ORANGE_6[@mode]
+
+        palette :fgDefault   => P_GRAY_9[@mode]
```
Comment (tancnle) [inline lib/rouge/themes/github.rb:79]: **suggestion:** Could we add a `bgDefault` palette for the background colour?

```ruby
P_WHITE = {:light => '#ffffff'}

...

palette :bgDefault   => P_WHITE[@mode]
```

**Review by tancnle — COMMENTED** [review]
Thank you for your contribution @dunkmann00 ❤️ The MR looks good overall. I have added some suggestions for your consideration. It would be nice to set the background colour correctly when viewing via the visual tool `http://localhost:9292/?theme=github.dark`

| Dark | Light |
| -- | -- |
| Screenshot 2023-01-24 at 9 43 42 pm | Screenshot 2023-01-24 at 9 43 29 pm |

 Please let me know what you think.

**Comment by dunkmann00** [comment]
Hi @tancnle! Thanks for taking a look at this and for offering some suggestions about the background. I've gone ahead and added a background color to the text for both modes.

**Comment by tancnle** [comment]
Thanks, @dunkmann00. Look awesome! 🚀

**Pull request merged**